cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs

```diff
@@ -0,0 +1,156 @@
+        // now `op` may be 'add' or 'sub'
+        // number is parsed with big endian
+
+        let splited = Vec::from_iter(msg.split("something").map(String::from));
+
+        let op = splited.get(0).unwrap();
+
+        match op.as_str() {
+            "add" => {
+                let op_num = splited.get(1).unwrap().parse::<u64>().unwrap();
+                if op_num != 2 {
+                    return Err(Error::from(ErrorKind::InvalidInputError));
+                }
+
+                let cipher_op1 = splited.get(2).unwrap();
+                let cipher_op2 = splited.get(3).unwrap();
+
+                self.input = Some(OperatorWorkerInput { 
+                    op: OperatorKind::Add,
+                    cipher_op1: cipher_op1.to_string(),
+                    cipher_op2: cipher_op2.to_string() });
+            },
+            "sub" => {
+                let op_num = splited.get(1).unwrap().parse::<u64>().unwrap();
+                if op_num != 2 {
+                    return Err(Error::from(ErrorKind::InvalidInputError));
+                }
+                let cipher_op1 = splited.get(2).unwrap();
+                let cipher_op2 = splited.get(3).unwrap();
+
+                self.input = Some(OperatorWorkerInput { op:OperatorKind::Sub,
+                    cipher_op1: cipher_op1.to_string(),
+                    cipher_op2: cipher_op2.to_string() });
+            },
+            _ => {
+                return Err(Error::from(ErrorKind::InvalidInputError));
+            }
+        }
+
+        Ok(())
+
+    }
+
+    fn execute(&mut self, _context: WorkerContext) -> Result<String> {
+        let input = self
+            .input
+            .take()
+            .ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
+
+        let key_pair = register_func::get_key_pair();
+        let s1 = vec![];
+        let s2 = vec![];
+        let op1 = eigen_crypto::ec::suite_b::ecies::decrypt(key_pair, &input.cipher_op1.as_bytes(), &s1, &s2).unwrap();
+        let op1 = u64::from_be_bytes(op1[0..8].try_into().unwrap());
+        let op2 = eigen_crypto::ec::suite_b::ecies::decrypt(key_pair, &input.cipher_op2.as_bytes(), &s1, &s2).unwrap();
+        let op2 = u64::from_be_bytes(op2[0..8].try_into().unwrap());
+
+        let result = match input.op {
+            OperatorKind::Add => {
+                op1 + op2
+            },
+            OperatorKind::Sub => {
+                op1 - op2
+            },
+            _ => {
+                return Err(Error::from(ErrorKind::InvalidInputError));
+            }
+        };
+
+        let s1 = vec![];
+        let s2 = vec![];
+        let key_pair = register_func::get_key_pair();
+
+        let alg = &eigen_crypto::sign::ecdsa::ECDSA_P256_SHA256_ASN1;
+        let public_key = eigen_crypto::sign::ecdsa::UnparsedPublicKey::new(alg, key_pair.public_key());
+
+        let cipher = eigen_crypto::ec::suite_b::ecies::encrypt(&public_key, &s1, &s2, &result.to_be_bytes());
```

Comment:
encrypt result by AES Key.

Reply:
OK. I've added these in the latest commitment.